packages/sdk/src/evm/common/sharded-merkle-tree.ts

```diff
@@ -73,17 +79,42 @@ export class ShardedMerkleTree {
     );
   }
 
-  static hashEntry(entry: SnapshotEntry, tokenDecimals: number): string {
-    return hashLeafNode(
-      entry.address,
-      utils.parseUnits(entry.maxClaimable, tokenDecimals),
-    );
+  static hashEntry(
+    entry: SnapshotEntry,
+    tokenDecimals: number,
+    snapshotFormatVersion: SnapshotFormatVersion,
+  ): string {
+    // TODO (cc) hash price and currency
+    // TODO (cc) differentiate between default values for maxClaimable
+
+    // TODO (cc) for legacy contracts, need to only hash addr / maxclaimable!!
+    switch (snapshotFormatVersion) {
+      case SnapshotFormatVersion.V1:
+        return utils.solidityKeccak256(
+          ["address", "uint256"],
+          [
+            entry.address,
+            convertQuantityToBigNumber(entry.maxClaimable, tokenDecimals),
+          ],
+        );
+      case SnapshotFormatVersion.V2:
+        return utils.solidityKeccak256(
+          ["address", "uint256", "uint256", "address"],
+          [
+            entry.address,
+            convertQuantityToBigNumber(entry.maxClaimable, tokenDecimals),
+            convertQuantityToBigNumber(entry.price, 18), // TODO (cc) use currencyDecimals
```

Comment:
I'm sure you remember this one but looks important so might as well call more attention to it (the TODO)

Reply:
Shouldn't be too bad to instantiate an IERC20 here to get the decimals

Reply:
yeah i do need to do that - thanks for the reminder. the async part is annoying that's why i didn't do it immediately